### PR TITLE
[tcgc] do not override client default value for api version param in non-versioning service

### PR DESCRIPTION
**Pull request by tadelesh**
fix: https://github.com/Azure/typespec-azure/issues/1608

### CODE REVIEW

File: packages/typespec-client-generator-core/test/packages/client.test.ts [hunk 1]
```diff
@@ -388,6 +388,73 @@ describe("typespec-client-generator-core: package", () => {
...
+    strictEqual(endpointTemplateArg.name, "endpoint");
+    strictEqual(endpointTemplateArg.onClient, true);
+    strictEqual(endpointTemplateArg.optional, false);
+    strictEqual(endpointTemplateArg.kind, "path");
+    strictEqual(endpointTemplateArg.clientDefaultValue, "http://localhost:3000");
+
+    const apiVersionParam = templatedEndpoint.templateArguments[1];
+    strictEqual(apiVersionParam.clientDefaultValue, "v1");
+    strictEqual(apiVersionParam.name, "apiVersion");
+    strictEqual(apiVersionParam.onClient, true);
```
Comment (haolingdong-msft) [inline packages/typespec-client-generator-core/test/packages/client.test.ts:453]: I'm a bit confused. why is non-versioning service still have `apiVersionParam.onClient ===  true`?

Comment (weidongxu-microsoft) [reply packages/typespec-client-generator-core/test/packages/client.test.ts:453]: It seems valid in this way, when the `api-version` is defined in `@server`.

I'd be interest to see a case of `api-version` defined in op (without versioning in service), and that parameter should have `onClient=false`.

**Review by weidongxu-microsoft — APPROVED** [review]

File: packages/typespec-client-generator-core/src/package.ts [hunk 2]
```diff
@@ -536,7 +536,7 @@ function getEndpointTypeFromSingleServer<
       }
       const apiVersionInfo = updateWithApiVersionInformation(context, param, client.__raw.type);
       sdkParam.isApiVersionParam = apiVersionInfo.isApiVersionParam;
-      if (sdkParam.isApiVersionParam) {
+      if (sdkParam.isApiVersionParam && apiVersionInfo.clientDefaultValue) {
```
Comment (chunyu3) [inline packages/typespec-client-generator-core/src/package.ts:539]: Is the `sdkParam.clientDefaultValue` always be override if apiVersionInfo.clientDefaultValue (if exist)? Is it possible that the sdkParam.clientDefaultValue has different value of  apiVersionInfo.clientDefaultValue, in that way , the sdkParam itself  default value is lost.

Considering the versioned service, and the `apiVersion` in the URL, with a default value (which may or may not the any enum value of the version enum)
Do we have UT to test this scenario?

Comment (tadelesh) [reply packages/typespec-client-generator-core/src/package.ts:539]: for versioning service, we honor the api version emitter specified, not client default value.

File: packages/typespec-client-generator-core/test/packages/client.test.ts (hunk 1 above)
Comment (haolingdong-msft) [reply packages/typespec-client-generator-core/test/packages/client.test.ts:453]: Could we add an unit test to test the case where api-version is not defined in `@service`, and `onClient` should be `false`?

Comment (tadelesh) [reply packages/typespec-client-generator-core/test/packages/client.test.ts:453]: we have an issue to track this: https://github.com/Azure/typespec-azure/issues/1595.

**Review by iscai-msft — APPROVED** [review]

**Pull request merged**